tests/unit/linking/test_link.py

```diff
@@ -189,3 +189,109 @@ def test_enhanced_match_three(self, session, enhanced_algorithm, patients: list[
+                        "system" : "http://terminology.hl7.org/CodeSystem/v2-0203",
+                        "code" : "SS"
+                    }]
+                },
+            }
+        ],
+        "name": [
+            {
+                "family": "Shepard",
+                "given": [
+                "John"
+                ],
+                "use": "official"
+            }
+        ],
+        "birthDate": "2053-11-07",
+        "gender": "male",
+        "address": [
+        {
+            "line": [
+            "1234 Silversun Strip"
+            ],
+            "buildingNumber": "1234",
+            "city": "Boston",
+            "state": "Massachusetts",
+            "postalCode": "99999",
+            "district": "county",
+            "use": "home"
+        }
+        ],
+        "telecom": [
+            {
+                "use": "home",
+                "system": "phone",
+                "value": "123-456-7890"
+            }
+        ],
+        "extension" : [
+            {
+                "url" : "http://hl7.org/fhir/StructureDefinition/individual-genderIdentity",
+                "extension" : [{
+                    "url" : "value",
+                    "valueCodeableConcept" : {
+                        "coding" : [{
+                            "system" : "http://snomed.info/sct",
+                            "code" : "446141000124107",
+                            "display" : "Identifies as female gender (finding)"
+                        }]
+                    }
+                }]
+            },
+            {
+                "url" : "http://hl7.org/fhir/us/core/StructureDefinition/us-core-race",
+                "extension": [
+                    {
+                        "url" : "ombCategory",
+                        "valueCoding" : {
+                            "system" : "urn:oid:2.16.840.1.113883.6.238",
+                            "code" : "2106-3",
+                            "display" : "White"
+                        }
+                    }
+                ]
+            }
+        ]
+    }
+
+    pii_record = link.fhir_record_to_pii_record(fhir_record)
+
+    assert pii_record.external_id == "f6a16ff7-4a31-11eb-be7b-8344edc8f36b"
+    assert pii_record.name[0].family == "Shepard"
+    assert pii_record.name[0].given == ["John"]
+    assert str(pii_record.birth_date) == "2053-11-07"
+    assert str(pii_record.sex) == "M"
+    assert pii_record.address[0].line == ["1234 Silversun Strip"]
+    assert pii_record.address[0].city == "Boston"
+    assert pii_record.address[0].state == "Massachusetts"
+    assert pii_record.address[0].postal_code == "99999"
+    assert pii_record.address[0].county == "county"
+    assert pii_record.mrn == "1234567890"
+    assert pii_record.ssn == "111-22-3333"
+    assert pii_record.telecom[0].value == "123-456-7890"
+    assert pii_record.telecom[0].system == "phone"
+    assert str(pii_record.race) == "WHITE"
+    assert str(pii_record.gender) == "FEMALE"
```

Comment:
This is correct, but I'm not seeing how you mapped "Identifies as female gender (finding)" to "FEMALE".  Does the translation happen in the `fhir_record_to_pii_record` function or in the pii schema?

Reply:
Oh I see, it tries to parse it as FHIR data, then falls back to check for tokens in the string.  However, for if NBS sends a payload like the following to the new link PII endpoint, will it result in a value of `Race.OTHER`?

```json
{
  "race": "Black"
}
```

Reply:
yes it would right now. I can add a couple if checks for some such as 
```
if val in ["black", "african American"]:
    return Race.BLACK
elif val in ["American indian", "alaska native"]:
    return Race.AMERICAN_INDIAN
...
...
...
return Race.OTHER
```

Reply:
I updated the gender parser to do something similar to the above in case the gender string isn't an exact match to the enum value

Reply:
So a little bit of context as we move away from FHIR and open this up to supporting more formats.

FHIR is quite verbose, and while it works to model our PII structure and enums around its definition, we'll likely have to be quite a bit more flexible when parsing data from other formats (eg NBS database, eCR, elr) as they won't be as verbose in their definitions.

I think some good test cases for these parsers would be to look at the spreadsheet that Krishna sent to us and check to see if the strings are parsed accordingly.  I think the one exception would be "race=Hispanic", I believe that should just be for ethnicity.

Reply:
Added some additional tests for the examples that Krishna sent over. And also added some if checks to `Gender` and `Race` to cover less verbose wording such as "black", "hawaiin" and other single word values.

Reply:
I guess I'm specifically worried about [this function](https://github.com/CDCgov/RecordLinker/blob/main/src/recordlinker/models/mpi.py#L100).  That is when we deserialize data from the database, are we able to reconstruct gender and race back into valid enums?  It's a little complicated so if you want to talk about this in detail later today I have time.

Reply:
Yeah that sounds good! I'd be down to talk about it later today to make sure it would work how we want it to

Reply:
I updated the [parse race](https://github.com/CDCgov/RecordLinker/pull/88/files#:~:text=def%20parse_race(cls%2C%20value)%3A) and parse gender functions and the corresponding enums to be non verbose and fhir based but still flexible